bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/api/IntesisBoxSocketApi.java

```diff
@@ -0,0 +1,232 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.intesis.internal.api;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.UnknownHostException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.thing.ThingStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class handling the Socket connections.
+ *
+ * @author Cody Cutrer - Initial contribution
+ * @author Hans-Jörg Merk - Moved Socket to it's own class
+ */
+@NonNullByDefault
+public class IntesisBoxSocketApi {
+
+    private final Logger logger = LoggerFactory.getLogger(IntesisBoxSocketApi.class);
+
+    private final String ipAddress;
+    private final int port;
+
+    private @Nullable IntesisSocket tcpSocket = null;
+    private @Nullable OutputStreamWriter tcpOutput = null;
+    private @Nullable BufferedReader tcpInput = null;
+
+    private @Nullable IntesisBoxChangeListener changeListener;
+
+    private boolean connected = false;
+
+    public IntesisBoxSocketApi(final String ipAddress, final int port) {
+        this.ipAddress = ipAddress;
+        this.port = port;
+    }
+
+    private class IntesisSocket {
+        final Socket socket;
+
+        public IntesisSocket() throws UnknownHostException, IOException {
+            socket = new Socket();
+            SocketAddress tcpSocketAddress = new InetSocketAddress(ipAddress, port);
+            socket.connect(tcpSocketAddress);
+        }
+
+        public void close() throws IOException {
+            socket.close();
+        }
+    }
+
+    @SuppressWarnings("null")
+    public void openConnection() throws IOException {
+        closeConnection();
+
+        logger.debug("openConnection(): Connecting to IntesisBox ");
+
+        IntesisBoxChangeListener listener = this.changeListener;
+        if (listener != null) {
+            logger.debug("IntesisBox listener added");
+        }
+
+        tcpSocket = new IntesisSocket();
+        tcpOutput = new OutputStreamWriter(tcpSocket.socket.getOutputStream(), "US-ASCII");
+        tcpInput = new BufferedReader(new InputStreamReader(tcpSocket.socket.getInputStream()));
+
+        Thread tcpListener = new Thread(new TCPListener());
+        tcpListener.start();
+
+        setConnected(true);
+        if (listener != null) {
+            listener.connectionStatusChanged(ThingStatus.ONLINE);
+        }
+    }
+
+    @SuppressWarnings("null")
+    public void closeConnection() {
+        try {
+            if (tcpSocket != null) {
```

Comment:
The field can be null at this point, because it isn't inittialized in the constructor. So, you need to check if it is not null before accessing it: if (job != null).But this would result in a compiler warning, as it is a field and could theoretically be modified by another thread exactly between the null check and the access in the next line.To mitigate this, you can store the field to a local variable before checking it:
```

Future<?> localPollingJob = pollingJob;
if (localPollingJob != null) {
    localPollingJob.cancel(true);
}
```

Reply:
Suppress Warning "null" is not needed any more.